platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/signed/SignedStateFileManager.java

```diff
@@ -297,6 +297,8 @@ private boolean saveSignedStateToDisk(
      * @param signedState the signed state to be written to disk.
      */
     public boolean saveSignedStateToDisk(final SignedState signedState) {
+        signedState.stateSavedToDisk();
```

Comment:
if this is where we set the value, probably its the same place we should check. cause there might be a race condition that results in the same state being saved twice still

Reply:
@lpetrovic05 I moved the check to this method

Reply:
Do we need to use a synchronized block or lock?   Thread 1 could be paused at line 314, and Thread 2 could enter the method and pass through 306-314.  Thread 1 wakes up and proceeds, both threads write the saved state to disk.    I think the update and check of the boolean variable need to be atomic.

Reply:
You're right - I'm thinking about what the best way to solve this is

Reply:
we can just do all the work inside the state writing thread. since its always a single thread, there will be no issues

Reply:
I moved the check and the set into the writing thread, as suggested @lpetrovic05